src/components/sync-connected-sites.tsx

```diff
@@ -208,9 +209,33 @@ const SyncConnectedSiteControls = ( {
 						type={ syncDialogType }
 						localSite={ selectedSite }
 						remoteSite={ connectedSite }
-						onSubmit={ () => {
+						onSubmit={ ( tree ) => {
+							const optionsToSync: SyncPart[] = [];
+
+							const isAll = tree.every( ( node ) => node.checked );
+							if ( isAll ) {
+								optionsToSync.push( 'all' );
+							} else {
+								const isDatabaseSelected = tree.find( ( node ) => node.id === 'sqls' )?.checked;
+
+								if ( isDatabaseSelected ) {
+									optionsToSync.push( 'sqls' );
+								}
+
+								const filesAndFolders =
+									tree.find( ( node ) => node.id === 'filesAndFolders' )?.children || [];
+								const wpContent =
+									filesAndFolders.find( ( node ) => node.id === 'wp-content' )?.children || [];
+
+								wpContent.forEach( ( item ) => {
+									if ( item.checked ) {
+										optionsToSync.push( item.id as SyncPart );
```

Comment:
I'm not fan of overriding types. Is there a way we can avoid forcing this `SyncPart` type?

Reply:
> I'm not fan of overriding types. Is there a way we can avoid forcing this SyncPart type?

I am also not a fan, and I would revert `optionsToSync` to `string`, it's one option.
The second option is to define `id`s with type SyncPart, BUT we have `id` as `filesAndFolders`, which actually is not allowed to send to backend, so I see two options:
1. use type assertion
2. return `optionsToSync ` to `string` type

Do you have otehr ideas?